src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    public static final double GROW_FACTOR = 1.5;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        this.elementData[size] = value;
+        size++;
     }
 
     @Override
```

Comment:
The 'grow' method should be refactored to include the size check within it. The name 'grow' suggests that the array will always grow, which is not the case. Consider renaming it to 'growIfNecessary' or 'ensureCapacity' to better reflect its functionality.